### PR TITLE
Adding Hyper and Http error variants. 

### PR DESCRIPTION
**Pull request by itsmeolivia**
Also,removes UnknownHashAlgorithm

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -54,7 +62,7 @@ pub enum Error {
 
     /// There is no known or available hash algorithm.
     #[error("unknown hash algorithm: {0}")]
-    UnkonwnHashAlgorithm(String),
+    UnknownHashAlgorithm(String),
```
Comment (erickt) [inline src/error.rs]: It seems like this isn't used. Could you just remove it?

File: src/error.rs [hunk 2]
```diff
@@ -26,6 +26,14 @@ pub enum Error {
     #[error("illegal argument: {0}")]
     IllegalArgument(String),
 
+    /// Generic error for HTTP connections.
+    #[error("http: {0}")]
+    Http(String),
```
Comment (erickt) [inline src/error.rs]: Why not propagate the error from the `http` crate?

File: src/error.rs [hunk 3]
```diff
@@ -26,6 +26,14 @@ pub enum Error {
...
+
+    /// Errors that can occur parsing HTTP streams.
+    #[error("hyper: {0}")]
+    Hyper(String),
```
Comment (erickt) [inline src/error.rs]: Here too.

**Review by erickt — COMMENTED** [review]
Since you are making a breaking change to the `Error` type, could you also mark this type as `#[non_exhaustive]`, as described by https://doc.rust-lang.org/1.35.0/unstable-book/language-features/non-exhaustive.html? This will allow us to add new error types in the future.

File: src/error.rs (hunk 2 above)
Comment (itsmeolivia) [reply src/error.rs]: http::Error and hyper::Error do not implement PartialEq or Eq

**Comment by itsmeolivia** [comment]
for the non_exhaustive tag, the ci needs to be updated to 1.40.0

**Comment by itsmeolivia** [comment]
closes #286

File: src/error.rs (hunk 2 above)
Comment (erickt) [reply src/error.rs]: Do we still need `PartialEq` and `Eq` if we're using `assert_matches`? Are there other places we need equality?

**Review by erickt — CHANGES_REQUESTED** [review]

**Pull request merged**